agrirouter-sdk-dotnet-standard-impl/Service/Common/EncodeMessageService.cs

```diff
@@ -39,6 +46,115 @@ public class EncodeMessageService
+                    var wholeMessage = messagePayloadParameters.Value.ToStringUtf8();
+                    var messageChunks = SplitByLength(wholeMessage,
+                        MessagePayloadParameters.MaxLengthForRawMessageContent).ToList();
+                    var messageParameterTuples = new List<MessageParameterTuple>();
+                    var chunkNr = 1;
+                    var chunkContextId = ChunkContextIdService.ChunkContextId();
+                    foreach (var messageChunk in messageChunks)
+                    {
+                        var messageId = MessageIdService.ApplicationMessageId();
+
+                        var chunkInfo = new ChunkComponent()
+                        {
+                            Current = chunkNr++,
+                            Total = messageChunks.Count(),
+                            ContextId = chunkContextId,
+                            TotalSize = wholeMessage.Length
+                        };
+
+                        var messageHeaderParametersForChunk = new MessageHeaderParameters()
+                        {
+                            Metadata = messageHeaderParameters.Metadata,
+                            Mode = messageHeaderParameters.Mode,
+                            Recipients = messageHeaderParameters.Recipients,
+                            ApplicationMessageId = messageId,
+                            TechnicalMessageType = messageHeaderParameters.TechnicalMessageType,
+                            TeamSetContextId = messageHeaderParameters.TeamSetContextId,
+                            ChunkInfo = chunkInfo
+                        };
+
+                        var messagePayloadParametersForChunk = new MessagePayloadParameters()
+                        {
+                            Value = ByteString.CopyFromUtf8(
+                                Convert.ToBase64String(Encoding.UTF8.GetBytes(messageChunk))),
+                            TypeUrl = messagePayloadParameters.TypeUrl,
+                        };
+
+                        messageParameterTuples.Add(new MessageParameterTuple
+                        {
+                            MessageHeaderParameters = messageHeaderParametersForChunk,
+                            MessagePayloadParameters = messagePayloadParametersForChunk
+                        });
+                    }
+
+                    return messageParameterTuples;
+                }
+                else
+                {
+                    Log.Debug("The message type needs to be base64 encoded, therefore we are encoding the raw value.");
+                    var messagePayloadParametersWithEncodedValue = new MessagePayloadParameters()
+                    {
+                        TypeUrl = messagePayloadParameters.TypeUrl,
+                        Value = ByteString.CopyFromUtf8(
+                            Convert.ToBase64String(messagePayloadParameters.Value.ToByteArray()))
```

Comment:
Not sure here: In all other parts of the code, ToStringUtf8(). Is there a special reason to use ToByteArray here? is the Result in all situations the same?

Reply:
Since the `Convert.ToBase64String` does take a `byte[]`, you can not use `ToStringUtf8` on its own. It would mean, that we first convert to a `string` and then to a `byte[]` which is quite redundant.

Reply:
I checked back the incident that I mentioned (127983 / 2019 in Launchpad) and it seems not to be relevant here.